### ISSUE TITLE
[MASSEMBLY-873] Maven-Assembly-Plugin freezes when building jar-with-dependencies of project depending on org.bouncycastle:bcprov-jdk15on:1.58

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#1086]
**[Michael Schierl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=schierlm)** opened **[MASSEMBLY-873](https://issues.apache.org/jira/browse/MASSEMBLY-873?redirect=false)** and commented

To reproduce:

1. Create a new directory
2. Add attached pom.xml (no other sources required to reproduce; in real world you would add some source that uses BouncyCastle lib, too).
3. Run `mvn package`

Actual result:

Maven hangs after

`[INFO] Building jar: C:\Daten\Eigenes\svn\iota\hang\target\bug-1.0-SNAPSHOT-jar-with-dependencies.jar`

, consuming one CPU core.

Expected result:

Maven builds a package containing no own source and the dependency (bouncycastle)

Workaround:

In maven local repository, edit the .jar file of BouncyCastle and remove the `META-INF\BC*` files. Then the package can be built again. So it seems to be caused somehow by verifying signed jar files.

---

**Affects:** 3.1.0

**Attachments:**
- [pom.xml](https://issues.apache.org/jira/secure/attachment/12903262/pom.xml) (_1.21 kB_)

**Issue Links:**
- [MASSEMBLY-875](https://issues.apache.org/jira/browse/MASSEMBLY-875) Maven Assembly Plugin 3.X is about 10x slower than 2.6
 (_**"duplicates"**_)
- [MASSEMBLY-900](https://issues.apache.org/jira/browse/MASSEMBLY-900) Upgrade plexus-archiver to 4.0.0
 (_**"depends upon"**_)

3 votes, 7 watchers

**Comment by jira-importer** [comment]
**[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz)** commented

New plexus-archiver released with fix - lets update with MASSEMBLY-900

**Comment by jira-importer** [comment]
**[Enrico Olivelli](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=eolivelli)** commented

 MASSEMBLY-900 landed to master.

[Michael Schierl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=schierlm) Do have have time to test your build with current master (3.1.1) ?

I will be happy to say that this issue is resolved in 3.1.1

**Comment by jira-importer** [comment]
**[Michael Schierl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=schierlm)** commented

[Enrico Olivelli](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=eolivelli) can you point me to a Maven repo that contains 3.1.1 (or a snapshot that I can test)? It does not seem to be in Maven Central yet.

 

```java
[ERROR] Plugin org.apache.maven.plugins:maven-assembly-plugin:3.1.1 or one of its dependencies could not be resolved: Could not find artifact org.apache.maven.plugins:maven-assembly-plugin:jar:3.1.1 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

 

Then I can test it with the project where the problem originally occurred.

**Comment by jira-importer** [comment]
**[Enrico Olivelli](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=eolivelli)** commented

You have to checkout from github and build it locally.

something like this:

```java
git clone http://github.com/apache/maven-assembly-plugin

mvn clean install -DskipTests
```

An you will find the 3.1.1-SNAPSHOT on your local repository

 

thank you

**Comment by jira-importer** [comment]
**[Michael Schierl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=schierlm)** commented

I can confirm that the issue is indeed fixed with the latest maven-assembly-plugin from GitHub.

**Comment by jira-importer** [comment]
**[Enrico Olivelli](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=eolivelli)** commented

Great! I will mark this issue as Resolved in 3.1.1.

I hope we will be able to release 3.1.1 in the beginning of January.
